### ISSUE TITLE
Populate grid subsets for faces and key points (like X-points, strike points)

### ISSUE THREAD

**Issue by eldond** [ProjectTorreyPines/SOLPS2IMAS.jl#6]
- [x] x_points: 6 (nodes defining X-points)
- [x] core_cut: 7 (y-aligned faces inside the separatrix connecting to the active x-point)
  - This tells us where the cut defined by b2fgmtry was
  - If extrapolated, this curve would pass through the magnetic axis
- [x] PFR_cut: 8 (y-aligned faces inside the private flux region connecting to the active x-point)
  - Another b2fgmtry cut
  - This is the boundary between inner and outer divertors
- [x] outer_throat: 9 (y-aligned faces in the outer SOL connecting to the active X-point)
  - This is the boundary between SOL and outer divertor
- [x] inner_throat: 10
- [x] outer_midplane: 11 (y-aligned faces connecting to the node closest to the outer midplane on the separatrix)
- [x] inner_midplane: 12
- [x] outer_target: 13 (y-aligned faces defining the outer target)
- [x] inner_target: 14
- [x] core_boundary: 15 (innermost x-aligned faces)
  - the inner edge of the mesh; boundary between computation region and blank spot in the core
- [x] separatrix: 16 (x-aligned faces defining the active separatrix)
- [x] outer_mid_plane_separatrix: 101 (point on active separatrix at outer midplane)
  - Intersection of 11 with 16
- [x] inner_mid_plane_separatrix: 102
  - Intersection of 12 with 16
- [x] outer_target_separatrix: 103 (point on active separatrix at outer [divertor] target)
  - Intersection of 13 with 16
- [x] inner_target_separatrix: 104
  - Intersection of 14 with 16

**Comment by anchal-physics** [comment]
https://github.com/ProjectTorreyPines/SOLPS2IMAS.jl/commit/a994b4d0150ad978468f0a204a003c7d5474b039 Added x_points

**Comment by anchal-physics** [comment]
https://github.com/ProjectTorreyPines/SOLPS2IMAS.jl/commit/8e399d098a38b90c2a2d29ea8c3ef4e2d6b4dd72 Added faces and core_cut

**Comment by anchal-physics** [comment]
https://github.com/ProjectTorreyPines/SOLPS2IMAS.jl/commit/760ef7b1b0bcab306d7d0d26f76f8f1a19545368 Added pfr_cut, throats, targets, fixed speed issue

**Comment by anchal-physics** [comment]
62371f2992c2071718452ebdd3ec6441451b632c Added inner and outer midplane subsets

**Comment by anchal-physics** [comment]
With new visualitation through GGDUtils, it has come to light that some subsets are not created properly:

![Image](https://github.com/ProjectTorreyPines/SOLPS2IMAS.jl/assets/29802951/362e109d-91f8-4157-999c-4a8537b86c14)
[SOLPSgridVisualization.pdf](https://github.com/ProjectTorreyPines/SOLPS2IMAS.jl/files/12654381/SOLPSgridVisualization.pdf)

Issues to fix:

- [ ] Inner throat edge is wrongly identified.
- [ ] Outer midplane are y-aligned edges but is that what we really wanted?
- [ ] Inner midplane did not get populated.

@eldond Please add more issues to fix if you can spot them.

**Comment by anchal-physics** [comment]
3062359a8c0ac111b55232fee215aa52af6f71c6 Inner throat edge issue fixed.



![Image](https://github.com/ProjectTorreyPines/SOLPS2IMAS.jl/assets/29802951/cc177acc-9065-4b7f-b3cc-f6088f2a1c05)

**Comment by anchal-physics** [comment]
3523eda569ede4f154f7e397386a19115b74b8c4



![Image](https://github.com/ProjectTorreyPines/SOLPS2IMAS.jl/assets/29802951/5046a941-3822-4df8-b506-1734dcb09362)

That was my bad. I got confused about y-aligned edge definition. I have corrected the outer midplane and inner midplane generation. The inner midplane would be populated if jxi information is provided in b2mn.dat. We can close this issue again now.
